File: modules/student-experience/src/main/java/com/campusos/portal/payload/InternshipPayload.java
package com.campusos.portal.payload;

import com.campusos.portal.domain.RequestType;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

public class InternshipPayload implements RequestPayload {

    /* ---- frozen contract fields ---- */
    public String company;
    public String role;
    public String from;
    public String to;
    public String details;
    public CertificateRef certificateRef;

    /* ---- system-derived ---- */
    public Sys sys = new Sys();

    public static class CertificateRef {
        public String filename;
        public String mime;
        public int sizeKb;

        public CertificateRef() {}

        public CertificateRef(String filename, String mime, int sizeKb) {
            this.filename = filename; this.mime = mime; this.sizeKb = sizeKb;
        }
    }

    public static class Sys {
        public int weeks;
        public String certificateCheck;
        public Integer credits;
        public String verifyId;
        public String documentSerial;
        public int returnCount;
    }

    @Override public RequestType type() { return RequestType.INTERNSHIP; }

    @Override public String title() { return role + " · " + company; }

    @Override public String subtitle() {
        return from + " → " + to + " · " + sys.weeks + " week(s)"
                + (certificateRef == null ? " · no certificate" : " · " + certificateRef.filename);
    }

    @Override public List<Artifact> artifacts() {
        List<Artifact> out = new ArrayList<>();
        if (sys.verifyId != null) {
            out.add(new Artifact("VERIFY_ID", "Verification ID", sys.verifyId, "/verify/" + sys.verifyId));
        }
        if (sys.credits != null) {
            out.add(Artifact.of("CREDITS", "Credits added to academic record", String.valueOf(sys.credits)));
        }
        if (sys.documentSerial != null) {
            out.add(Artifact.of("SERIAL", "Certificate published as", sys.documentSerial));
        }
        return out;
    }

    @Override public void validate() {
        if (company == null || company.isBlank()) throw new IllegalArgumentException("company is required");
        if (role == null || role.isBlank()) throw new IllegalArgumentException("role is required");
        LocalDate a = LocalDate.parse(from), b = LocalDate.parse(to);
        if (!b.isAfter(a)) throw new IllegalArgumentException("end date must be after start date");
        if (b.isAfter(LocalDate.now())) throw new IllegalArgumentException("internship has not ended yet");
    }
}

File: modules/student-experience-js/src/payload/InternshipPayload.js
import { RequestType } from '../domain/enums.js';
import { parseDate, IllegalArgumentException } from './RequestPayload.js';
import { artifact } from './Artifact.js';

export class InternshipPayload {
  constructor() {
    this.company = null;
    this.role = null;
    this.from = null;
    this.to = null;
    this.details = null;
    this.certificateRef = null; // { filename, mime, sizeKb }
    this.sys = { weeks: 0, certificateCheck: null, credits: null, verifyId: null,
      documentSerial: null, returnCount: 0 };
  }

  type() { return RequestType.INTERNSHIP; }

  title() { return `${this.role} · ${this.company}`; }

  subtitle() {
    return `${this.from} → ${this.to} · ${this.sys.weeks} week(s)`
      + (this.certificateRef === null ? ' · no certificate' : ` · ${this.certificateRef.filename}`);
  }

  artifacts() {
    const out = [];
    if (this.sys.verifyId !== null) {
      out.push(artifact('VERIFY_ID', 'Verification ID', this.sys.verifyId, `/verify/${this.sys.verifyId}`));
    }
    if (this.sys.credits !== null) {
      out.push(artifact('CREDITS', 'Credits added to academic record', String(this.sys.credits)));
    }
    if (this.sys.documentSerial !== null) {
      out.push(artifact('SERIAL', 'Certificate published as', this.sys.documentSerial));
    }
    return out;
  }

  handledBy() { return null; }

  validate() {
    if (!this.company || !this.company.trim()) throw new IllegalArgumentException('company is required');
    if (!this.role || !this.role.trim()) throw new IllegalArgumentException('role is required');
    const a = parseDate('start date', this.from);
    const b = parseDate('end date', this.to);
    if (!(b.getTime() > a.getTime())) throw new IllegalArgumentException('end date must be after start date');
    if (b.getTime() > Date.now()) throw new IllegalArgumentException('internship has not ended yet');
  }

  static fromJSON(o) { return Object.assign(new InternshipPayload(), o, { sys: { ...new InternshipPayload().sys, ...o.sys } }); }
}
